pkg/controllers/proxy/network_services_controller.go

```diff
@@ -286,16 +295,23 @@ func (nsc *NetworkServicesController) Run(healthChan chan<- *healthcheck.Control
 	// https://github.com/kubernetes/kubernetes/pull/70530/files
 	setSysCtlAndCheckError(utils.IPv4ConfAllArpAnnounce, arpAnnounceUseBestLocalAddress)
 
-	// Ensure rp_filter=2 for DSR capability, see:
+	// Ensure rp_filter=2 (or leave 0 untouched) for DSR capability, see:
 	// * https://access.redhat.com/solutions/53031
 	// * https://github.com/cloudnativelabs/kube-router/pull/1651#issuecomment-2072851683
+	// Only override rp_filter if it is set to 1, as enabling it from 0 to 2 can cause issues
+	// with some network configurations which use reverse routing. All must be overriden as it overrides others
+	rpFilterAll := checkRpFilter1("all")
 	if nsc.krNode.IsIPv4Capable() {
-		sysctlErr := utils.SetSysctlSingleTemplate(utils.IPv4ConfRPFilterTemplate, "all", 2)
-		if sysctlErr != nil {
-			if sysctlErr.IsFatal() {
-				klog.Fatal(sysctlErr.Error())
-			} else {
-				klog.Error(sysctlErr.Error())
+		for _, ifname := range []string{"kube-bridge", "kube-dummy-if", nsc.krNode.GetNodeInterfaceName()} {
+			if rpFilterAll || checkRpFilter1(ifname) {
```

Comment:
Just setting the `rp_filter` setting on the main interfaces is most likely enough, but I've had enough experience with this project where making smaller changes breaks some sort of edge case for someone else. So since I'm likely to miss some edge case for someone else in my testing, I'd feel better if we could just add the `all` interface here to this list instead of the check above.

My thought is that without setting `rp_filter` on the `all` interface, we'll miss the `veth` devices that are created by the CNI dynamically. We'll also miss other people's interfaces tunnelling interfaces for things like wireguard if they have those or some such.

Thoughts?

Reply:
veth might be the case, yes. So perhaps if all is 1 we set it to 2, if 0 we keep it

Reply:
Yeah, I agree.

Reply:
Just wanted to check in, did you want to make this change? Or would you prefer for me to push to your branch with the change?

Reply:
Ah, will do in a moment. Got caught into other tasks briefly. Sorry for a delay